flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClientFactory.java

```diff
@@ -77,6 +78,8 @@ public ApplicationId getClusterId(Configuration configuration) {
     private YarnClusterDescriptor getClusterDescriptor(Configuration configuration) {
         final YarnClient yarnClient = YarnClient.createYarnClient();
         final YarnConfiguration yarnConfiguration = new YarnConfiguration();
+        yarnConfiguration.addResource(HadoopUtils.getHadoopConfiguration(configuration));
```

Comment:
Why is it necessary to add the `HadoopConfiguration` here?

Reply:
I find this `yarnConfiguration` can be used to get FileSystem in YarnClusterDescriptor.class, such as in`startAppMaster` method:
![image](https://user-images.githubusercontent.com/13013780/103879642-172d3800-5113-11eb-860e-fe5f1b7bbc45.png)
So add `HadoopConfiguration` to benefit the filesystem.

Reply:
ok, thanks for the clarification.